cirq-core/cirq/ops/linear_combinations.py

```diff
@@ -498,6 +498,23 @@ def _unitary_(self) -> np.ndarray:
             return m
         raise ValueError(f'{self} is not unitary')
 
+    def _json_dict_(self):
+        def key_json(k):
+            # k is a frozenset, each element of frozen set is a tuple.
+            return [[x for x in e] for e in sorted(list(k))]
+
+        return {
+            'keys': [key_json(k) for k in self._linear_dict.keys()],
+            'values': list(self._linear_dict.values()),
+        }
+
+    @classmethod
+    def _from_json_dict_(cls, keys, values, **kwargs):
+        mapping = dict()
+        for k, v in zip(keys, values):
+            mapping[frozenset(tuple(x) for x in k)] = v
```

Comment:
you can use a dict comprehension here

```python
{frozenset(tuple(x) for x in k): v for k, v in zip(keys, values)}
```

Reply:
can we use slightly more helpful names than `k` `x` and `e`? Maybe a scheme where their length relates to their degree of nestiness?